exonum/src/blockchain/mod.rs

```diff
@@ -82,8 +82,8 @@ pub struct Blockchain {
 
 impl Blockchain {
     /// Constructs a blockchain for the given `storage` and list of `services`.
-    pub fn new(
-        storage: Box<Database>,
+    pub fn new<D: Into<Arc<Database>>>(
+        storage: D,
```

Comment:
Technically, this is a breaking change, right?

Reply:
I do not think so. In old code you can pass `Box::new(MemoryDb::new))` but now you need to convert `MemoryDb` into `Arc<Database>` manually or just pass `MemoryDb::new` I you do not need customization.